### PR TITLE
[ADAPTEDS-91] Edit profile functionality

### PR DESCRIPTION
**Pull request by kyperbelt**
[Jira](https://terabite-adapted-strength.atlassian.net/browse/ADAPTEDS-91)

### Description
The application user needs a place to edit some of the profile information. 

### Solution
Created a new front end page at the `/edit-profile` route that has a few fields to allow users to edit basic information like `first_name`, `last_name`, `phone`, `address`. This sends a request to the backend `v1/user/profile` endpoint and updates the information in the database.

### Testing
Does the project build? 
- **Yes** both the backend and the frontend build successfully and operate as intended. 

What else did you do to test this? 
- Just ran a few edge cases manually and fixed up some additional things. 

### Other Changes In This PR
* **Added** a new script to the `scripts/` folder called `check_env_vars.sh` that automatically runs when trying to deploy spring boot backend. This is to warn you that your environment variables are not yet set. Simply set them to something. 
    > :warning: **YOU MUST SET ADAPTED_STRENGTH_WEB_URL to `localhost` OTHWERWISE you will not be able to test locally** :warning: 

* **Refactored** the backend so that what would have originally been their own microservice is now in their own package. This is to reduce coupling between Apis in the event that we have time to split this into microservices in the future. Currently there is only the `authorization` and `user` packages, but there is room to grow this. See joshuas architecture diagram for an idea of what packages we might need to create. 

* **Added** :information_source: Frontend :information_source:  a new `api` folder that contains the `ApiUtils.js`, `AuthApi.js` and `UserApi.js` classes. They should allow us to easily hook in to the backend when need be. For example, to get the user information is as simple as calling 
    ```javascript
     import {UserApi} from '.api/UserApi';
     ...
     UserApi.getUserInformation().then(response => {
           const data = response.data;
           // do something with the data here
    }); 
     ...
    ```
* **Added** :information_source: Frontend :information_source: new React Components called `<RouteGuard>` and `<StateGuard>`. 
    - **StateGuard**: This allows you to control the rendering of a react component based on some given state without having to do a `useState` setup. 
    - **RouteGuard**: This is mainly to be used with Router routes, but it allows you to set a state function that decides if you render the route or redirect to an alternative route. This is used right now to redirect from pages that are meant for signed in users.
        For Example, the following will redirect the route to the login page if the user is not logged in, otherwise it will display the EditProfile page.
        ```javascript 
               <RouteGuard state={AuthApi.isLoggedIn} routeTo="/login">
                     <EditProfile />
              </RouteGuard>

        ```
* **Integrated** the backend and the frontend to some extent. There might be some work that needs to be redone, but for now we have something we can demo to the client tomorrow at 8pm. 
    - You should now be able to do the following: 
        - Login 
        - Logout 
        - Edit Profile
     - Other frontend pages are still not hooked up like the `Signup`, but with the added utilities and examples I think we should be able to hook those up with ease in the coming sprint.

### CODE REVIEW

File: authorization/src/main/java/com/terabite/user/controller/UpdateInformationRequestBody.java [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+import com.fasterxml.jackson.annotation.JsonAlias;
+
+/**
+ * This class is used to map the request body of the update information request.
+ * Its different because we dont want to allow updating all the fields after
+ * initial account setup.
+ * For example, the user should not be able to change their email address after
+ * the account is created since that is the unique identifier for the account.
+ */
+public class UpdateInformationRequestBody {
```
Comment (jcupler) [inline authorization/src/main/java/com/terabite/user/controller/UpdateInformationRequestBody.java:12]: Consider using a `record` class

**Review by jcupler — APPROVED** [review]
Question: What is the logger that LoggerFactory is using, and where does it write the logs to.
Otherwise: Java looks good - it might have to be changed or expanded as the design and backed services grow, but it's at a good place right now, I think. I know nothing of JS, so consider waiting for someone on the front end to review.
